src/selectors/entities/channels.test.js

```diff
@@ -9,6 +9,884 @@ import {sortChannelsByDisplayName, getDirectChannelName} from 'utils/channel_uti
 import * as Selectors from 'selectors/entities/channels';
 import {General, Preferences} from '../../constants';
 
+const sortUsernames = (a, b) => a.localeCompare(b, General.DEFAULT_LOCALE, {numeric: true});
+
+describe('Selectors.Channels.getChannelsInCurrentTeam', () => {
+    const team1 = TestHelper.fakeTeamWithId();
+    const team2 = TestHelper.fakeTeamWithId();
+
+    it('should return channels in current team', () => {
+        const user = TestHelper.fakeUserWithId();
+
+        const profiles = {};
+        profiles[user.id] = user;
+
+        const channel1 = TestHelper.fakeChannelWithId(team1.id);
+        channel1.display_name = 'Channel Name';
+        const channel2 = TestHelper.fakeChannelWithId(team2.id);
+        channel2.display_name = 'Channel Name';
+        const channel3 = TestHelper.fakeChannelWithId(team1.id);
+        channel3.display_name = 'Channel Name';
+        const channel4 = TestHelper.fakeChannelWithId('');
+        channel4.display_name = 'Channel Name';
+
+        const channels = {};
+        channels[channel1.id] = channel1;
+        channels[channel2.id] = channel2;
+        channels[channel3.id] = channel3;
+        channels[channel4.id] = channel4;
+
+        const channelsInTeam = {};
+        channelsInTeam[team1.id] = [channel1.id, channel3.id];
+        channelsInTeam[team2.id] = [channel2.id];
+        channelsInTeam[''] = [channel4.id];
+
+        const testState = deepFreezeAndThrowOnMutation({
+            entities: {
+                users: {
+                    currentUserId: user.id,
+                    profiles,
+                },
+                teams: {
+                    currentTeamId: team1.id,
+                },
+                channels: {
+                    channels,
+                    channelsInTeam,
+                },
+            },
+        });
+
+        const channelsInCurrentTeam = [channel1, channel3].sort(sortChannelsByDisplayName.bind(null, []));
+        assert.deepEqual(Selectors.getChannelsInCurrentTeam(testState), channelsInCurrentTeam);
+    });
+
+    it('should order by user locale', () => {
```

Comment:
Good to finally get a test for these. I know locale can affect sorting, but I didn't know of any cases where that happened before.

Reply:
All merit to the [function documentation](https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Objetos_globales/String/localeCompare). I got the example they had there and used it.